src/provider/provider_tracking.c

```diff
@@ -80,6 +80,8 @@ static umf_result_t umfMemoryTrackerRemove(umf_memory_tracker_handle_t hTracker,
 
 umf_memory_pool_handle_t umfMemoryTrackerGetPool(const void *ptr) {
     assert(ptr);
+    assert(TRACKER);
```

Comment:
I don;t know if we should assert it. Creating memory tracker can fail for a few reasons, see here: https://github.com/oneapi-src/unified-memory-framework/blob/f2b784afcc54b360e46e0a098c7f3b2641b31625/src/provider/provider_tracking.c#L464. I think we should rather just log this.

Reply:
Done

Reply:
Replaced with:
```c
    if (TRACKER == NULL) {
        fprintf(stderr, "tracker is not created\n");
        return NULL;
    }

    if (TRACKER->map == NULL) {
        fprintf(stderr, "tracker's map is not created\n");
        return NULL;
    }
```

Done